File: cryptography/caesar-cipher/caesar-cipher.c
#include <stdio.h>

#define	MAX_TEXT_LENGTH	50

void caesar_cipher_encrypt (char *cipher_text, const char *const plain_text, int key) {
	int i = 0;
	for (; i < MAX_TEXT_LENGTH && plain_text[i] != '\0'; i++) {
		if (plain_text[i] >= 'A' && plain_text[i] <= 'Z') {
			// right shifting key positions for A-Z characters
			cipher_text[i] = 'A' + (plain_text[i] - 'A' + key) % 26;
		} else if (plain_text[i] >= 'a' && plain_text[i] <= 'z') {
			// right shifting key positions for a-z characters
			cipher_text[i] = (plain_text[i] - 'a' + key) % 26;
		} else {
			cipher_text[i] = plain_text[i];
		}
	}
	if (i < MAX_TEXT_LENGTH) {
		cipher_text[i] = '\0';
	}
}

void caesar_cipher_decrypt (char *decrypted_message, const char *const cipher_text, int key) {
	// decrypting by left shifing by key positions
	caesar_cipher_encrypt(decrypted_message, cipher_text, -key);
}

int main () {
	char plain_text[MAX_TEXT_LENGTH] = "BRUTUS IS A TRAITOR!";
	char cipher_text[MAX_TEXT_LENGTH], decrypted_message[MAX_TEXT_LENGTH];
	int key = 5;

	caesar_cipher_encrypt(cipher_text, plain_text, key);
	caesar_cipher_decrypt(decrypted_message, cipher_text, key);

	printf("Plain text: %s\n", plain_text);
	printf("Encrypted cipher text: %s\n", cipher_text);
	printf("Decrypted message: %s\n", decrypted_message);

	return 0;
}

File: cryptography/caesar-cipher/caesar-cipher.py
def caesar_cipher_encrypt (plain_text: str, key: int) -> str:
    cipher_text = ''
    for c in plain_text:
        if 'A' <= c <= 'Z':
            # right shifting key positions for A-Z characters
            cipher_text += chr(ord('A') + (ord(c) - ord('A') + key) % 26)
        elif 'a' <= c <= 'z':
			# right shifting key positions for a-z characters
            cipher_text += chr(ord('a') + (ord(c) - ord('a') + key) % 26)
        else:
            cipher_text += c

    return cipher_text


def caesar_cipher_decrypt (cipher_text: str, key: int) -> str:
	# decrypting by left shifing by key positions
    return caesar_cipher_encrypt(cipher_text, -key);


def main ():
    plain_text = "BRUTUS IS A TRAITOR!"
    key = 5

    cipher_text = caesar_cipher_encrypt(plain_text, key)
    decrypted_message = caesar_cipher_decrypt(cipher_text, key)

    print(f'Plain text: {plain_text}')
    print(f'Encrypted cipher text: {cipher_text}')
    print(f'Decrypted message: {decrypted_message}')


if __name__ == '__main__':
    main()
